docs/content/preview/explore/ysql-language-features/data-manipulation.md

```diff
@@ -1,386 +1,371 @@
 
 ```sql
-COPY employees FROM '/home/mydir/employees.txt.sql' DELIMITER ',' CSV HEADER;
+UPDATE employees SET department = 'Brand Management' WHERE department = 'Marketing';
 ```
 
-## Export data to a file
-
-The `COPY TO` statement allows you to export data from a table to a file. By specifying a column list, you can instruct `COPY TO` to only export data from certain columns.
+To update the department of all employees to `Sales`, you can run
 
-The `filename` parameter of the `COPY` statement enables copying to a file directly.
+```sql
+UPDATE employees SET department = 'Sales';
+```
 
-The following example demonstrates how to use the `COPY FROM` statement:
+You can also perform mathematical operations on columns where permitted. For example, you can change the ids for all employees by incrementing by 1 like:
 
 ```sql
-COPY employees TO '/home/mydir/employees.txt.sql' DELIMITER ',';
+UPDATE employees SET id = id + 1;
 ```
 
-## Back up a database
-
-You can back up a single instance of a YugabyteDB database into a plain-text SQL file by using the  `ysql_dump` is a utility, as follows:
+## Upsert
 
-```sh
-ysql_dump mydb > mydb.sql
-```
+An `UPSERT` is a combination of "UPDATE" and "INSERT," allowing you to either insert a new row into a table or update an existing row if it already exists. This operation is particularly useful for avoiding duplication while ensuring data is either added or updated as needed. This functionality is provided by the `INSERT ... ON CONFLICT DO UPDATE` clause.
 
-To back up global objects that are common to all databases in a cluster, such as roles, you need to use  `ysql_dumpall` .
+For example, if we try to insert record `(1, 'Johnny Depp', 'Sales')` you will get an error like:
 
-`ysql_dump` makes backups regardless of whether or not the database is being used.
+```sql{.nocopy}
+ERROR:  23505: duplicate key value violates unique constraint "employees_pkey"
+```
 
-To reconstruct the database from a plain-text SQL file to the state the database was in at the time of saving, import this file using the `\i` meta-command, as follows:
+This is because, the record `(1, 'Johnny Depp', 'Marketing')` already exists. Now instead of an error to be thrown, we can either ask the server to ignore the insert with `DO NOTHING` like:
 
 ```sql
-yugabyte=# \i mydb
+INSERT INTO employees (id, name, department)
+    VALUES (1, 'Johnny Depp', 'Sales')
+    ON CONFLICT (id)
+    DO NOTHING;
 ```
 
-## Define NOT NULL constraint
-
-YSQL lets you define various constraints on columns. One of these constraints is [NOT NULL](../indexes-constraints/other-constraints/#not-null-constraint). You can use it to specify that a column value cannot be null. Typically, you apply this constraint when creating a table, as the following example demonstrates:
+which would return `INSERT 0 0` to signify no rows were inserted, or we can choose to update the any other column other than the conflicted column (say, department) like:
```

Comment:
```suggestion
This would return `INSERT 0 0` to signify no rows were inserted. Alternatively, you can choose to update any other column other than the conflicted column (say, department) like so:
```